### PR TITLE
Update kotlinx-metadata to 0.7.0

### PR DESCRIPTION
**Pull request by ZacSweers**
Following the guidance here: https://github.com/JetBrains/kotlin/blob/master/libraries/kotlinx-metadata/jvm/Migration.md#migrating-from-06x-to-070

This removes a lot of our preview helper APIs as most of them are part of the first party APIs now

### CODE REVIEW

File: interop/kotlinx-metadata/api/kotlinx-metadata.api [hunk 1]
```diff
@@ -1,97 +1,3 @@
-public final class com/squareup/kotlinpoet/metadata/FlagsKt {
```
Comment (Egorand) [inline interop/kotlinx-metadata/api/kotlinx-metadata.api:1]: Is it worth it being super diligent and deprecating all of these with `@ReplaceWith`?

File: interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ElementsClassInspector.kt [hunk 2]
```diff
@@ -333,8 +332,7 @@ public class ElementsClassInspector private constructor(
 
     val propertyData = declarationContainer.properties
       .asSequence()
-      .filter { it.isDeclaration }
-      .filterNot { it.isSynthesized }
```
Comment (Egorand) [inline interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ElementsClassInspector.kt:337]: No need for this `filterNot` anymore?

File: interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ReflectiveClassInspector.kt [hunk 3]
```diff
@@ -319,8 +318,7 @@ public class ReflectiveClassInspector private constructor(
 
     val propertyData = declarationContainer.properties
       .asSequence()
-      .filter { it.isDeclaration }
-      .filterNot { it.isSynthesized }
```
Comment (Egorand) [inline interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ReflectiveClassInspector.kt:323]: Same here.

**Review by Egorand — APPROVED** [review]

File: interop/kotlinx-metadata/api/kotlinx-metadata.api (hunk 1 above)
Comment (ZacSweers) [reply interop/kotlinx-metadata/api/kotlinx-metadata.api:1]: I don't think so, especially as many of them are just import changes

File: interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ElementsClassInspector.kt (hunk 2 above)
Comment (ZacSweers) [reply interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ElementsClassInspector.kt:337]: Nope, one thing learned in this is that this is mutually exclusive with declaration/delegation/etc

Comment (Egorand) [reply interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/classinspectors/ElementsClassInspector.kt:337]: Nice

File: interop/kotlinx-metadata/api/kotlinx-metadata.api (hunk 1 above)
Comment (Egorand) [reply interop/kotlinx-metadata/api/kotlinx-metadata.api:1]: Might make sense to call out the deletions in the release notes.

**Pull request merged**